include/time_series/internal/base.hxx

```diff
@@ -213,3 +223,20 @@ size_t TimeSeriesBase<P, T>::max_length()
     read_indexes();
     return this->history_elements_ptr_->size();
 }
+
+template <typename P, typename T>
+bool TimeSeriesBase<P, T>::is_empty()
+{
+    if (!empty_)
```

Comment:
Just for understanding: Why is `empty_` used instead of simply checking `newest_timeindex_ < oldest_timeindex_`?

Reply:
because it saves the "read_indexes" operation